ssl/record/rec_layer_s3.c

```diff
@@ -869,27 +871,31 @@ int do_ssl3_write(SSL *s, int type, const unsigned char *buf,
             SSL3_RECORD_add_length(thiswr, 1);
 
             /* Add TLS1.3 padding */
+            rlen = SSL3_RECORD_get_length(thiswr);
+            max_padding = SSL3_RT_MAX_PLAIN_LENGTH - rlen;
             if (s->record_padding_cb != NULL) {
-                size_t rlen = SSL3_RECORD_get_length(thiswr);
-
                 padding = s->record_padding_cb(s, type, rlen, s->record_padding_arg);
                 /* do not allow the record to exceed max plaintext length */
-                if (padding > (SSL3_RT_MAX_PLAIN_LENGTH - rlen))
-                    padding = SSL3_RT_MAX_PLAIN_LENGTH - rlen;
+                if (padding > max_padding)
+                    padding = max_padding;
             } else if (s->block_padding > 0) {
                 size_t mask = s->block_padding - 1;
                 size_t remainder;
 
                 /* optimize for power of 2 */
                 if ((s->block_padding & mask) == 0)
-                    remainder = SSL3_RECORD_get_length(thiswr) & mask;
+                    remainder = rlen & mask;
                 else
-                    remainder = SSL3_RECORD_get_length(thiswr) % s->block_padding;
+                    remainder = rlen % s->block_padding;
                 /* don't want to add a block of padding if we don't have to */
-                if (remainder == 0)
+                if (remainder == 0) {
                     padding = 0;
-                else
+                } else {
                     padding = s->block_padding - remainder;
+                    /* do not allow the record to exceed max plaintext length */
+                    if (padding > max_padding)
+                        padding = max_padding;
+                }
             }
             if (padding > 0) {
                 if (!WPACKET_memset(thispkt, 0, padding)) {
```

Comment:
I thought of that; but if |max_padding| is zero, there's no point in doing the WPACKET_memset(), etc. So that case is filtered out by the `padding > 0` check